### ISSUE TITLE
this reference does not throw if argument to super

### ISSUE THREAD

**Issue by monsanto** [babel/babel#1284]
```js
class X extends Foo {
    constructor() {
        super(this.x)
    }
}
```

This compiles fine; I expect it to be rejected with `'this' is not allowed before super()`

**Comment by sebmck** [comment]
Fixed as of 5.1.11, thanks!

**Comment by nesukun** [comment]
This change breaks working, existing, albeit invalid code. I think it should be documented so projects can update their code before it breaks their tests.
